File: problemas/iniciante/1153/1153.py
def Fatorial(n):
    if(F[n] == -1):
        F[n] = Fatorial(n - 1) * n
    return F[n]


F = [-1 for _ in range(14)]
F[0] = 1

N = int(input())

print(Fatorial(N))

File: problemas/iniciante/1153/1153.r
F <- rep(-1, 13)
F[1] = 1

fatorial <- function(x){
    if(F[x] == -1){
        F[x] = x * fatorial(x - 1)
    }
    return (F[x])
}

input <- file('stdin', 'r')

N <- as.integer(readLines(input, n=1))
write(paste(fatorial(N)), '')